request/graphql/schema/generate.go

```diff
@@ -194,8 +194,12 @@ func (g *Generator) generate(ctx context.Context, collections []client.Collectio
 		var isReadOnly bool
 		var collectionFound bool
 		for _, definition := range collections {
-			if t.Name() == definition.Description.Name {
-				isReadOnly = definition.Description.BaseQuery != nil
+			if t.Name() == definition.Description.Name.Value() {
```

Comment:
> question: Why have you skipped `HasValue()` here before calling `Value()`. If IRRC the value returned is invalid unless `HasValue() == true`

question: Same as above

Reply:
Any collection in the gql types very definitely has a name - this is it's reason to be.